azurerm/resource_arm_storage_blob.go

```diff
@@ -149,6 +156,7 @@ func resourceArmStorageBlobCreate(d *schema.ResourceData, meta interface{}) erro
 	blobType := d.Get("type").(string)
 	cont := d.Get("storage_container_name").(string)
 	sourceUri := d.Get("source_uri").(string)
+	contentType := d.Get("content_type").(string)
 
 	log.Printf("[INFO] Creating blob %q in storage account %q", name, storageAccountName)
 	if sourceUri != "" {
```

Comment:
Refer to this article to simplify the `happy-path` code block:
https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88

Reply:
we can pull this out into separate methods - but I think that's outside of the scope of this PR tbh; so let's do that in a separate PR

Reply:
Agree with @tombuildsstuff . The blob storage code is a messy and I also think the fields are not good and make people confused. That will be a big refactoring, so let's do it in another PR if possible.

Reply:
also, at _some point in the future_ (read: when the SDK's are more stable) we'll be switching over to the new [Storage SDKs](http://github.com/Azure/azure-storage-blob-go/issues/48); which would probably be a sensible time to do this refactoring